packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -27,11 +27,12 @@ abstract class Notification {
   @protected
   @mustCallSuper
   bool visitAncestor(Element element) {
-    if (element is StatelessElement &&
-        element.widget is NotificationListener<Notification>) {
-      final NotificationListener<Notification> widget = element.widget;
-      if (widget._dispatch(this, element)) // that function checks the type dynamically
-        return false;
+    if (element is StatelessElement) {
+      StatelessWidget widget = element.widget;
+      if (widget is NotificationListener<Notification>) {
+        if (widget._dispatch(this, element)) // that function checks the type dynamically
```

Comment:
The reason the is check on the left doesn't result in inferring that widget is a NotificationListener<Notification>
is that element.widget could change between when the is check is performed and when it is accessed.
Adding a local variable earlier fixes that issue.